File: backend/assetflow-auth-service/src/main/java/com/assetflow/auth/entity/Department.java
package com.assetflow.auth.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "departments")
@Getter
@Setter
@NoArgsConstructor
public class Department {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "department_id")
	private Long id;

	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "company_id")
	private Company company;

	@Column(name = "department_name")
	private String name;

	private String description;

	@Column(name = "is_active")
	private boolean active = true;
}

File: dotNet/AuthService/Models/Department.cs
namespace AuthService.Models;

public class Department
{
    public long Id { get; set; }
    public long? CompanyId { get; set; }
    public Company? Company { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public bool Active { get; set; } = true;
}
